k8s/client_test.go

```diff
@@ -0,0 +1,144 @@
+package k8s
+
+import (
+	"testing"
+
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInvalidUrlReturnsError(t *testing.T) {
+	_, err := New("%gh&%ij", []byte{}, "")
+	assert.Error(t, err)
+}
+
+func TestRetrievesIngressesFromKubernetes(t *testing.T) {
+	assert := assert.New(t)
+
+	ingressFixture := CreateIngressesFixture()
+	ts := httptest.NewTLSServer(handleGetIngresses(t, authToken, ingressFixture))
+	defer ts.Close()
+
+	client, err := New(ts.URL, caCert, authToken)
+	assert.NoError(err)
```

Comment:
This oddly returns a boolean rather than failing the test. I tend to use `assert.Nil` on errros.